frontend/src/views/admin/matching/index.tsx

```diff
@@ -21,11 +83,218 @@ export function AdminMatchingView() {
+    const positionSummaries = React.useMemo(() => {
+        const assignmentsByPositionId: Record<number, Assignment[]> = {};
+        for (const assignment of assignments) {
+            assignmentsByPositionId[assignment.position.id] =
+                assignmentsByPositionId[assignment.position.id] || [];
+            assignmentsByPositionId[assignment.position.id].push(assignment);
+        }
+
+        const applicantSummariesByPositionId: Record<
+            number,
+            ApplicantSummary[]
+        > = {};
+        for (const applicant of applicants) {
+            const newestApplication = GetNewestApplication(
+                applicant,
+                applications
+            );
+            if (!newestApplication) {
+                continue;
+            }
+
+            const applicantMatches =
+                matches.filter((match) => match.applicantId === applicant.id) ||
+                [];
+
+            const applicantGuarantee =
+                guarantees.find(
+                    (guarantee) => guarantee.utorid === applicant.utorid
+                ) || null;
+
+            const newApplicantSummary = {
+                applicant: applicant,
+                mostRecentApplication: newestApplication,
+                matches: applicantMatches,
+                guarantee: applicantGuarantee,
+            };
+
+            newApplicantSummary.mostRecentApplication.position_preferences.forEach(
+                (position) => {
+                    applicantSummariesByPositionId[position.position.id] =
+                        applicantSummariesByPositionId[position.position.id] ||
+                        [];
+                    applicantSummariesByPositionId[position.position.id].push(
+                        newApplicantSummary
+                    );
+                }
+            );
+
+            // Add summary to positions where the applicant has been assigned:
+            assignments
+                .filter(
+                    (assignment) => assignment.applicant.id === applicant.id
+                )
+                .forEach((assignment) => {
+                    applicantSummariesByPositionId[assignment.position.id] =
+                        applicantSummariesByPositionId[
+                            assignment.position.id
+                        ] || [];
+
+                    const existingSummary = applicantSummariesByPositionId[
+                        assignment.position.id
+                    ].find((summary) => summary.applicant.id === applicant.id);
+
+                    // Add the applicant summary to the position only if the summary does not already exist
+                    if (!existingSummary) {
+                        applicantSummariesByPositionId[
+                            assignment.position.id
+                        ].push(newApplicantSummary);
+                    }
+                });
+        }
+
+        const ret: Record<number, PositionSummary> = {};
```

Comment:
What?? How/why are you indexing by a number? All ids are strings...

Ah, I see this is the Rails id, which actually is a number. Give `ret` a better name.